bibernate-core/src/main/java/com/bobocode/petros/bibernate/session/jdbc/DefaultJdbcQueryManager.java

```diff
@@ -1,46 +1,162 @@
+        Objects.requireNonNull(restrictions, "Parameter [restrictions] must not be null!");
+        final String sql = QueryBuilder.buildQuery(type, QueryType.SELECT, restrictions);
+        try (final Connection connection = this.dataSource.getConnection();
+             final PreparedStatement statement = StatementUtils.prepareSelectStatement(connection.prepareStatement(sql), type, restrictions)) {
+            final ResultSet resultSet = statement.executeQuery();
+            final QueryResult queryResult = EntityUtils.getQueryResult(resultSet, type);
+            final C result = queryResult.isSingle() ? queryResult.wrap(queryResult.getSingleResult()) :
+                    queryResult.isMulti() ? queryResult.getResultList() : (C) Collections.emptyList();
+            log.debug("Query: '{}', result: {}", sql, result);
+            return result;
+        } catch (SQLException e) {
+            throw new JdbcOperationException(e.getMessage(), e);
+        }
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param entity entity that is supposed to be updated
+     * @param <T>    generic type
+     * @return updated entity
+     */
     @Override
-    public <T> T update(T entity) {
-        return null;
+    public <T> T update(final T entity) {
+        Objects.requireNonNull(entity, "Parameter [entity] must not be null!");
+        final String sql = QueryBuilder.buildQuery(entity.getClass(), QueryType.UPDATE, Collections.emptyList());
+        try (final Connection connection = this.dataSource.getConnection();
+             final PreparedStatement statement = StatementUtils.prepareUpdateStatement(connection.prepareStatement(sql), entity)) {
+            final int rowsUpdated = statement.executeUpdate();
+            log.debug("Updated '{}' rows for query {}", rowsUpdated, sql);
+            return entity;
+        } catch (SQLException e) {
+            throw new JdbcOperationException(e.getMessage(), e);
+        }
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param entity entity that is supposed to be deleted
+     * @param <T>    generic type
+     */
     @Override
-    public <ID> void deleteById(ID id) {
-
+    public <T> void delete(final T entity) {
+        Objects.requireNonNull(entity, "Parameter [entity] must not be null!");
+        try {
+            final Field idField = EntityUtils.getIdField(entity.getClass());
+            idField.setAccessible(true);
+            deleteById(entity.getClass(), idField.get(entity));
+        } catch (IllegalAccessException e) {
+            throw new JdbcOperationException(e.getMessage(), e);
+        }
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param type entity type
+     * @param id   id (primary key)
+     * @param <T>  generic type
+     * @param <ID> primary key type of the given entity
+     */
     @Override
-    public <T, ID> void delete(Class<T> type, ID id) {
-
+    public <T, ID> void deleteById(final Class<T> type, final ID id) {
```

Comment:
Looks like generic parameter ID is redundant here and we can use a simple Object here